core/capabilities/remote/utils.go

```diff
@@ -93,3 +100,39 @@ func AggregateModeRaw(elemList [][]byte, minIdenticalResponses uint32) ([]byte,
 	}
 	return found, nil
 }
+
+func SanitizeLogString(s string) string {
+	tooLongSuffix := ""
+	if len(s) > maxLoggedStringLen {
+		s = s[:maxLoggedStringLen]
+		tooLongSuffix = " [TRUNCATED]"
+	}
+	for i := 0; i < len(s); i++ {
+		if !unicode.IsPrint(rune(s[i])) {
+			return "[UNPRINTABLE] " + hex.EncodeToString([]byte(s)) + tooLongSuffix
+		}
+	}
+	return s + tooLongSuffix
+}
+
+// Workflow IDs and Execution IDs are 32-byte hex-encoded strings
+func IsValidWorkflowOrExecutionID(id string) bool {
+	if len(id) != validWorkflowIDLen {
+		return false
+	}
+	_, err := hex.DecodeString(id)
+	return err == nil
+}
+
+// Trigger event IDs and message IDs can only contain printable characters and must be non-empty
```

Comment:
Out of curiosity, is there a reason why we can't have the same ID validation as above? Namely, could Trigger event IDs and message IDs also be 32-byte hex-encoded strings?

Reply:
Currently trigger event IDs are human readable, e.g. "streams_123456890" and I think that makes it debug-friendly. As for Message IDs we could probably impose some extra restriction ...